src/imu/imu.cpp

```diff
@@ -0,0 +1,242 @@
+#include <iostream>
+#include <cmath>
+#include <numeric>
+
+#include <robottestingframework/TestAssert.h>
+#include <robottestingframework/dll/Plugin.h>
+
+#include <yarp/os/Property.h>
+#include <yarp/os/Stamp.h>
+
+#include "imu.h"
+
+using namespace robottestingframework;
+ROBOTTESTINGFRAMEWORK_PREPARE_PLUGIN(Imu)
+
+Imu::Imu() : TestCase("Imu") { }
+
+Imu::~Imu() { }
+
+bool Imu::setup(yarp::os::Property& property) {
+    
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("robot"), "The robot name must be given as the test parameter!");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("port"), "The port name must be given as the test parameter!");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("model"), "Please, provide the urdf model path.");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("frame"), "Please, provide the frame name.");
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(property.check("sensor"), "Please, provide the sensor name.");
+    
+    robotName = property.find("robot").asString(); // robot name
+    portName = property.find("port").asString(); // name of the port from which the data are streamed
+    modelPath = property.find("model").asString(); // urdf model path
```

Comment:
We typically rely on the YARP way of retrieving parameters to automatically load the correct URDF based on the currently set `YARP_ROBOT_NAME`. To do so, you can do:

```suggestion
    modelName = property.find("model").asString(); // urdf model path
    yarp::os::ResourceFinder &rf = yarp::os::ResourceFinder::getResourceFinderSingleton();
    std::string modelAbsolutePath = rf.findFileByName(suggestion);
```

See https://github.com/robotology/whole-body-estimators/blob/5fb9aeeb1cc9282189eee11754c78f685814ed9f/devices/baseEstimatorV1/src/fbeRobotInterface.cpp#L96 for an example of this used. In that way, `model` can just be `model.urdf` and you always load the right one. See also https://github.com/robotology/icub-models#use-the-models-from-c-using-yarp .